main/src/cgeo/geocaching/ui/dialog/CoordinatesCalculateDialog.java

```diff
@@ -0,0 +1,1096 @@
+                    resortEquations();
+                    updateResult();
+                }
+            });
+        }
+
+        return v;
+    }
+
+    // Make this dialog completely fill the screen
+    @Override
+    public void onStart() {
+        super.onStart();
+        final Dialog d = getDialog();
+        if (d != null) {
+            d.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
+        }
+    }
+
+    // Save the current state of the calculator such that it can be restored after screen rotation (or similar).
+    @Override
+    public void onSaveInstanceState(final Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putByteArray("calc_state", SerializationUtils.serialize(saveState()));
+    }
+
+    private void displayToast(final int message) {
+        displayToast(message, false);
+    }
+
+    private void displayToast(final int message, final boolean shortToast) {
+        final AbstractActivity activity = (AbstractActivity) getActivity();
+        if (shortToast) {
+            activity.showShortToast(activity.getString(message));
+        } else {
+            activity.showToast(activity.getString(message));
+        }
+    }
+
+    // This method has largely been lifted from the CoordinateInputDialog and is used to validate
+    // coordinate specifications prior to assigning them to the Geopoint.
+    private boolean areCurrentCoordinatesValid(final boolean signalError) {
+        // then convert text to geopoint
+        final Geopoint current;
+        final String lat = getLatResult();
+        final String lon = getLonResult();
+
+        final String[] latTokens = lat.split("[ °'\"\\.]+");
+        final String[] lonTokens = lon.split("[ °'\"\\.]+");
+
+        try {
+            switch (currentFormat) {
+                case Deg:
+                    current = new Geopoint(latTokens[0], latTokens[1], latTokens[2], lonTokens[0], lonTokens[1], lonTokens[2]);
+                    break;
+                case Min:
+                    current = new Geopoint(latTokens[0], latTokens[1], latTokens[2], latTokens[3], lonTokens[0], lonTokens[1], lonTokens[2], lonTokens[3]);
+                    break;
+                case Sec:
+                    current = new Geopoint(latTokens[0], latTokens[1], latTokens[2], latTokens[3], latTokens[4], lonTokens[0], lonTokens[1], lonTokens[2], lonTokens[3], lonTokens[4]);
+                    break;
+                case Plain:
+                    current = new Geopoint(lat, lon);
+                    break;
+                default:
+                    throw new IllegalStateException("can never happen, keep tools happy");
+            }
+            if (current.isValid()) {
+                gp = current;
+                return true;
+            }
+        } catch (final Geopoint.ParseException ignored) {
+            // Signaled and returned below
+        }
+
+        if (signalError) {

```

Comment:
That's how it was done in`CoordinatesInputDialog` so I just kind of copied it.

Anyway, it looks like it's a flag to indicate whether the error message should be displayed as a toast or not.  As it is, in `CoordinatesCalculateDialog` it appears to only be called with a 'true' parameter so I'll just get rid of the variable.